pkg/cloudprovider/cloudprovider.go

```diff
@@ -82,15 +86,18 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 	subnetProvider := NewSubnetProvider(ec2api)
 	instanceTypeProvider := NewInstanceTypeProvider(ctx, ctx.Session, ec2api, subnetProvider, ctx.UnavailableOfferingsCache, ctx.StartAsync)
+	amiProvider := amifamily.NewAmiProvider(ctx.KubeClient, ssm.New(ctx.Session), ec2api, cache.New(awscontext.CacheTTL, awscontext.CacheCleanupInterval), cache.New(awscontext.CacheTTL, awscontext.CacheCleanupInterval), ctx.KubernetesInterface)
```

Comment:
Hey @jonathan-innis , I have refactored amiProvider out and it now has `GetAMIsForProvider`.

Reply:
Moved `KubeServerVersion(ctx)` to amiProvider too, I think it belongs there only. As only for ami we need that for now.